scripts/install.sh

```diff
@@ -0,0 +1,224 @@
+#!/bin/bash
+
+ROOT_DIR="$(pwd)"
+
+if [[ "$OSTYPE" == "darwin"* ]]; then
+    OS="macOS"
+    NUMBER_OF_PROCESSORS="$(sysctl -n hw.ncpu)"
+fi
+
+if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+    OS="Linux"
+    NUMBER_OF_PROCESSORS="$(nproc)"
+fi
+
+GENERATOR="Unix Makefiles"
+
+SUBMODULE_DIR="${ROOT_DIR}/vendor"
+SUBMODULE_INCLUDE_DIR="${SUBMODULE_DIR}/include"
+SUBMODULE_LIB_DIR="${ROOT_DIR}/lib/${OS}"
+
+echo "Installing vulkan dependencies..."
+
+update_submodules() {
+    git submodule update --init ${SUBMODULE_DIR}/$1
+}
+
+mkdirp() { 
+    mkdir -p $1 
+}
+
+setup_volk() {
+    echo "Cloning Volk..."
+    update_submodules volk
+
+    mkdirp ${SUBMODULE_DIR}/include/volk
+
+    cp ${SUBMODULE_DIR}/volk/volk.h ${SUBMODULE_INCLUDE_DIR}/volk 
+    cp ${SUBMODULE_DIR}/volk/volk.c ${SUBMODULE_INCLUDE_DIR}/volk
+}
+
+setup_glfw() {
+    echo "Cloning glfw..."
+
+    update_submodules glfw
+
+    cd ${SUBMODULE_DIR}/glfw ; cmake -G "${GENERATOR}" . ; make -j${NUMBER_OF_PROCESSORS}
+
+    mkdirp ${SUBMODULE_LIB_DIR}
+
+    cp ${SUBMODULE_DIR}/glfw/src/libglfw3.a ${SUBMODULE_LIB_DIR}
+
+    cd ${ROOT_DIR}
+}
+
+setup_moltenVk() {
+    echo "Cloning MoltenVk..."
+
+    update_submodules MoltenVK
+    cd ${SUBMODULE_DIR}/MoltenVK ; ./fetchDependencies --macos --v-headers-root ${SUBMODULE_DIR}/Vulkan-Headers
```

Comment:
can we not just run:
```suggestion
    ./${SUBMODULE_DIR}/MoltenVK/fetchDependencies --macos --v-headers-root ${SUBMODULE_DIR}/Vulkan-Headers
```

Reply:
Done. I've changed most of the instances where we run CD. In the case of the code above, it looks like the script relies on relative paths. I've wrapped the code in parentheses to open a subshell and ensure we don't end up in a random directory.